### ISSUE TITLE
Setting up new desc-pipelines conda-forge metapackage

### ISSUE THREAD

**Issue by heather999** [LSSTDESC/desc-help#47]
As discussed on Slack: https://lsstc.slack.com/archives/G8BV3R04C/p1603276418046700
Plan is to create a new desc-pipelines metapackage in conda-forge.  Need to collect a full list of expected packages and looking for input!
Expectation is that the package list will be based on what is already in [desc-python](https://github.com/LSSTDESC/desc-python/blob/master/conda/desc-python-env.yml)
Note some of these packages may only available via pip:

- https://github.com/LSSTDESC/CatalogMatcher/archive/master.tar.gz
- ceci
- https://github.com/LSSTDESC/descqa/archive/v2.0.0-0.7.0.tar.gz
- fast3tree
- fits2hdf
- https://github.com/yymao/FoFCatalogMatching/archive/v0.1.0.tar.gz
- gcr
- gcr-catalogs
- https://bitbucket.org/yymao/helpers/get/v0.3.2.tar.gz
- https://github.com/msimet/Stile/archive/v0.1.tar.gz

 May want to add some additional packages:

- [parallel_statistics](https://github.com/LSSTDESC/parallel_statistics/)
- [POT](https://pythonot.github.io/)

Other suggestions?

**Comment by beckermr** [comment]
I'm happy to work on conda-forge packages for many of these things. A few notes. 

1. The packages need to have released versions with version numbers. 

2. Their dependencies will need to be in conda-forge. The items in this list apply to dependencies as well naturally. 

3. We may want to consider using a prefix in the package names to help distinguish desc software. See eg the current ones we have (https://github.com/conda-forge?q=lsstdesc&type=&language=).

4. All packages will need a license that allows redistribution (e.g, BSD-3-Clause, GPL, MIT, etc) and be public ofc.

Some of the authors may need to go back and address these items.

Bump (@yymao, @fjaviersanchez, and others I am missing)

**Comment by erykoff** [comment]
@beckermr are we going to have science pipelines v19 (or 20?) in conda-forge (via stackvana?). If so, I can put `suprême` in conda-forge.  Probably with a DESC prefix.

**Comment by beckermr** [comment]
I have not pushed `stackvana` but I can. I was waiting for @brianv0 and co to finish their work on the DM side of the house. I have version 2020.40 of `stackvana` built here (https://anaconda.org/beckermr/stackvana/files) for linux and OSX. 

I am more than happy to review PRs for any conda-forge package by DESC members, so do feel free to make a PR when you are ready. Submit them here (https://github.com/conda-forge/staged-recipes) and bump me when you are ready.

**Comment by beckermr** [comment]
Note that some packages may not want a desc prefix. For example, we did not add one for CCL and adding one for things like `ceci` doesn't make a ton of sense. OTOH DESC-specific code that is unlikely to be used outside the collaboration may benefit from having such a prefix.

**Comment by heather999** [comment]
Some packages may have names that are too generic. It doesn't have to be a desc prefix, but some packages may want to ponder their future name... like `helpers`, if that is meant to be a long-term supported package.

**Comment by jmeyers314** [comment]
If it's easy to `pip install blah --user`, then that's sufficient.  If not, then I request `ipdb` and `ptpython`.

**Comment by yymao** [comment]
Hmm, I am not sure if I have the time/resources to make my packages listed above into conda-forge. Are they all required by this new desc-pipelines metapackage?

**Comment by beckermr** [comment]
@yymao I am happy to make conda-forge packages for whatever we'd like. I don't want to go around releasing versions though. 

We don't need to move all of them. even having a rough metapackage in conda-forge would help DESC users standup environments in various places that are close to what is at nersc.

**Comment by yymao** [comment]
Thanks, @beckermr.  In Heather's list, packages that I am the primary maintainer of all have versioned releases, and all have open source licenses (MIT, and in one case GPL), so I think they already satisfy Matt's (1) and (4). Below I list their dependency situation: 

### General-purpose packages
- [fast3tree](https://bitbucket.org/yymao/fast3tree): all dependencies are in conda-forge; installation requires building C code. 
- [FoFCatalogMatching](https://github.com/yymao/FoFCatalogMatching): with the exception of `fast3tree` above, all other dependencies are in conda-forge.
- [easyquery](https://github.com/yymao/easyquery): all dependencies are in conda-forge.
- [gcr](https://github.com/yymao/generic-catalog-reader): with the exception of `easyquery` above, all other dependencies are in conda-forge.
- [helpers](https://bitbucket.org/yymao/helpers): with the exception of `fast3tree` above, all other dependencies are in conda-forge. Note that *no* long-term support was intended for this package.

### DESC-specific packages
- [gcr-catalogs](https://github.com/LSSTDESC/gcr-catalogs): with the exception of `gcr` above, all other dependencies are in conda-forge.
- [descqa](https://github.com/LSSTDESC/descqa): all core dependencies are all in conda-forge, but some additional dependencies (e.g., `GCR`, `CatalogMatcher`) are not.

**Comment by fjaviersanchez** [comment]
Thanks for the ping @beckermr! I added a license and tagged [CatalogMatcher](https://github.com/LSSTDESC/CatalogMatcher). This package depends on `numpy`, `scipy` and `scikit-learn` which I think are all in conda-forge. The package is of general purpose but I'm happy to add the `lsstdesc` prefix. Please let me know if there's anything else I should do in order to make things easier.

**Comment by heather999** [comment]
Hi @beckermr Sorry for the long silence.  Using the [rubinenv-feedstock](https://github.com/conda-forge/rubinenv-feedstock) example and the general instructions in the conda-forge doc - I went ahead and forked conda-forge/staged-recipes, made a branch and added a first stab at a recipe for a lsstdesc-env, based on what we have in desc-python and leaving off what is just available via pip.  I tried a local build and nothing horrible happend! I might be brave enough to open a PR and see where this goes :) https://github.com/heather999/staged-recipes/tree/lsstdesc-env/recipes/lsstdesc.env
I followed the naming convention for rubinenv where it looks like the actual installable package is `rubin-env` but the feedstock is named `rubinenv`.  Am I in the ballpark?  I figure things can be fixed in PR, but if there are obvious things to change might as well address them now.

**Comment by beckermr** [comment]
Looks great. One minor comment

We should remove this line 

```
  skip: True         # [python_impl == 'pypy' or win]
```

and add

```
noarch: generic
```

to the build section.

This will make the recipe installable on any platform. If some of the deps are not available, then it simply won't install which is fine.

As far as the package name, the rubin folks made a mistake when making the feedstock. I'd make both lsstdesc-env.

We should also decide on a versioning scheme for the env now.

**Comment by beckermr** [comment]
We could also go with `lsstdesc.env` (i.e. a period instead of a dash) to match the package naming convention.

**Comment by heather999** [comment]
While I like the dash, I'm happy to stick with the `lsstdesc.env` so it matches the other `lsstdesc.` packages. Now updated!

Open to suggestions on the versioning scheme.. I was just going with  major.minor.patch and starting with 0.Y.Z for now - since this is all preliminary.  That was about as far as I thought about it :)

One question, I noticed rubinenv included their own [conda_build_config.yaml ](https://github.com/conda-forge/rubinenv-feedstock/blob/master/recipe/conda_build_config.yaml) and found: https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7 but also https://conda-forge.org/docs/maintainer/infrastructure.html#centos-sysroot-for-linux-platforms 
Is it beneficial to do the same for lsstdesc.env? Is it irrelevant since the default currently seems to be the same? I haven't given enough thought to supporting a build environment, but there will be those that may use the env for building development versions. Are there other considerations to help support that better?

**Comment by beckermr** [comment]
> Open to suggestions on the versioning scheme.. I was just going with major.minor.patch and starting with 0.Y.Z for now - since this is all preliminary. That was about as far as I thought about it :)

Right. So usually people use semver to set API expectations. Is that the case here? Do we want to use a version that is a straight date?

> One question, I noticed rubinenv included their own conda_build_config.yaml and found: https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7 but also https://conda-forge.org/docs/maintainer/infrastructure.html#centos-sysroot-for-linux-platforms
Is it beneficial to do the same for lsstdesc.env? Is it irrelevant since the default currently seems to be the same? I haven't given enough thought to supporting a build environment, but there will be those that may use the env for building development versions. Are there other considerations to help support that better?

You do not need this. This was done to support centos7 system libs for parts of the DM stack. We won't need that or we will let the stack pull that in itself.

**Comment by beckermr** [comment]
Another thing. We can trade the dot for a dash in package names now if you'd like. I got the dot from @rmjarvis and lsstdesc.coord. I wonder if he has options here.

**Comment by heather999** [comment]
For versioning - lsstdesc.env itself has no API but some of the included packages will - does that need to be expressed?  Dates will make it abundantly clear how old/new a version is, but I wonder if it will be easy to refer to down the road..  would people easily recall something like 2020.01.14 (or 2020.01.14_micro if there's some chance there is a quick patch on the same day)?  I'm always worried about confusing colleagues with dates if it's unclear what is referring to the month and what is the day.. but I think YYYY.MM.DD is pretty common.

I do prefer the dash, but willing to yield to other opinions on that.  I figure the most important thing is consistency.

**Comment by beckermr** [comment]
*opinions

**Comment by beckermr** [comment]
> For versioning - lsstdesc.env itself has no API but some of the included packages will - does that need to be expressed? 

Right, so if we were to increase a major or minor version each time an underlying package was updated, we'd be pushing them all the time. I prefer a date system myself. Also remember this is an abstract metapackage. It specifies what to install but not necessarily which versions. That is left to conda and the underlying constraints of the packages themselves. So we shouldn't worry too much.

**Comment by rmjarvis** [comment]
What are you asking for opinions about?
LSSTDESC.Coord is mostly just to make it more specific that merely Coord, which was (and still is) available, but seemed presumptuous of me to claim for this package, so I wanted something a little longer.  I don't think I have a preference of names with dot vs hyphen.  Either works.

**Comment by beckermr** [comment]
We were asking about `.` vs `-`. Thanks!

**Comment by erykoff** [comment]
I personally prefer the `-` to the `.`.  Certainly it's much more common.

**Comment by beckermr** [comment]
KK @heather999 let's go with a hyphen. I will change the other packages we have to use both in order to keep things backwards compatible.

**Comment by heather999** [comment]
Ok - updated to use the hyphen!  As for version, sure let's go with dates:  YYYY.MM.DD   Any need to initially indicate a `_patch` number?  I usually prefer consistency, so rather than introducing the `_patch` as needed, I'd include it by default.. so `2020.11.10_0`

**Comment by beckermr** [comment]
For this, I am not too worried about patch numbers. The use of this env since it contains abstract specs is to help people easily bootstrap an env with the right packages locally. Others should weigh in here. I know what I would like but other people might have better ideas.

**Comment by heather999** [comment]
Ok - let's move forward, how about `2020.11.11` :)  new day.. and we can go from there?  Once that's set, I'll submit a PR and we can see what happens next!

**Comment by beckermr** [comment]
Yup. See the package conda-forge/conda-forge-pinning-feedstock. We can autogenerate the version.

**Comment by beckermr** [comment]
We could also append the build number to the version.

**Comment by heather999** [comment]
ah yes, autogenerating the version - I was wondering about that!  Ok - let's see if I can get this in.  I like the idea of appending the build number too - I'll just out and ask how to do that, is that stored in `PKG_BUILDNUM`?

**Comment by beckermr** [comment]
make a jinja2 var and then add it to the version. I can make the change on staged recipes if you make a PR there.

**Comment by heather999** [comment]
Great, thanks!  I submitted the [PR](https://github.com/conda-forge/staged-recipes/pull/13207#issuecomment-725852534) and see the failure for win64.  Just digging around, starting with `fitsio` which is listed as one of the failed packages, I see the fitsio conda-forge build [skips win](https://github.com/conda-forge/fitsio-feedstock/blob/69cf1307d492cbfeb913d6e2b97a41f85a41d215/recipe/meta.yaml#L16) same situation with [`namaster`](https://github.com/conda-forge/namaster-feedstock/blob/b73fb5a84799d42476660cce22e71389330a20bb/recipe/meta.yaml#L15)
So at least for now, I think lsstdesc-env will need to skip win too - agreed?

**Comment by beckermr** [comment]
noarch recpes only build on linux. You can ignore the windows failure.

**Comment by heather999** [comment]
Well that's done!  So now we sit back and suddenly `lsstdesc-env` will become available to install? :)
Next up, we work through the remaining packages needing to become conda-forged?  I might play with `easy-query` since it  looks.. easy...

**Comment by beckermr** [comment]
We don't have to rename the repos. We can make the conda name different than the repo name.

**Comment by yymao** [comment]
I've made [easyquery](https://github.com/yymao/easyquery) into conda-forge (https://github.com/conda-forge/easyquery-feedstock) :tada:.  Since easyquery was already in PyPI, [grayskull](https://github.com/conda-incubator/grayskull) makes it easy to create the recipe.  The conda-forge community is also very helpful. 

Now that easyquery is in, I think [GCR](https://github.com/yymao/generic-catalog-reader) can be done relatively straightforward as well, since it's also in PyPI and we can use grayskull. For [fast3tree](https://bitbucket.org/yymao/fast3tree), while it is in PyPI, the C library part may makes it not as straightforward. Other packages in my [list above](https://github.com/LSSTDESC/desc-help/issues/47#issuecomment-714820622) are not in PyPI and I have not learned how to generate recipes for them yet.

**Comment by beckermr** [comment]
And yes the conda-forge community is a really nice one.

**Comment by beckermr** [comment]
@yymao I opened an issue on fast3tree to help guide changes to the code base to make the installation more robust and also easier for conda: https://github.com/yymao/fast3tree/issues/3 

We will likely need to address this issue first before we can properly build it in conda-forge.

**Comment by beckermr** [comment]
For recipes not in pypi, you can pull the source from the github release but you will need to write the meta.yaml by hand. I can start in on those.

**Comment by yymao** [comment]
Thanks @beckermr! When I was tidying up the fast3tree repo yesterday, I had a feeling what you said in https://github.com/yymao/fast3tree/issues/3 would be an issue.  There's however a complicating factor about how the so files are generated...  I'll ask you some questions on Slack about this later.

**Comment by heather999** [comment]
@beckermr A question just came up from @cwwalter about the ability to search for all DESC packages available in conda-forge.  We discussed using that `lsstdesc-` prefix, but not all DESC packages already in conda-forge have that prefix.  Do we want to encourage the use of this prefix more universally?  Are there cases where it is inappropriate?  Chris specifically asked if there a way to at least tag all packages as desc so in coda-forge?

**Comment by beckermr** [comment]
We decided that for DESC specific packages, we would use the prefix, but that for more general things, we would not. You can use `conda search lsstdesc` to find things

```
$ conda search lsstdesc
Loading channels: done
No match found for: lsstdesc. Search: *lsstdesc*
# Name                       Version           Build  Channel             
lsstdesc-env            2020.11.12.0      hd8ed1ab_0  conda-forge         
lsstdesc.coord                 1.0.5 py27h04f5b5a_1000  conda-forge         
lsstdesc.coord                 1.0.5  py27h2d50403_0  conda-forge         
lsstdesc.coord                 1.0.5  py35h2d50403_0  conda-forge         
lsstdesc.coord                 1.0.5 py36h04f5b5a_1000  conda-forge         
lsstdesc.coord                 1.0.5  py36h2d50403_0  conda-forge         
lsstdesc.coord                 1.0.5 py37h04f5b5a_1000  conda-forge         
lsstdesc.coord                 1.0.5  py37h2d50403_0  conda-forge         
lsstdesc.coord                 1.1.0  py27h04f5b5a_0  conda-forge         
lsstdesc.coord                 1.1.0  py36h04f5b5a_0  conda-forge         
lsstdesc.coord                 1.1.0  py37h04f5b5a_0  conda-forge         
lsstdesc.coord                 1.1.1  py27h770b8ee_0  conda-forge         
lsstdesc.coord                 1.1.1  py36h770b8ee_0  conda-forge         
lsstdesc.coord                 1.1.1  py37h770b8ee_0  conda-forge         
lsstdesc.coord                 1.1.2  py27h770b8ee_0  conda-forge         
lsstdesc.coord                 1.1.2  py36h770b8ee_0  conda-forge         
lsstdesc.coord                 1.1.2  py37h770b8ee_0  conda-forge         
lsstdesc.coord                 1.2.0  py27h770b8ee_0  conda-forge         
lsstdesc.coord                 1.2.0  py27ha1b3eb9_0  conda-forge         
lsstdesc.coord                 1.2.0  py36h06ab557_1  conda-forge         
lsstdesc.coord                 1.2.0  py36h770b8ee_0  conda-forge         
lsstdesc.coord                 1.2.0  py36h863e41a_1  conda-forge         
lsstdesc.coord                 1.2.0  py36ha1b3eb9_0  conda-forge         
lsstdesc.coord                 1.2.0  py37h770b8ee_0  conda-forge         
lsstdesc.coord                 1.2.0  py37ha1b3eb9_0  conda-forge         
lsstdesc.coord                 1.2.0  py37ha1cc60f_1  conda-forge         
lsstdesc.coord                 1.2.0  py38ha0d09dd_1  conda-forge         
lsstdesc.coord                 1.2.0  py38ha1b3eb9_0  conda-forge         
lsstdesc.coord                 1.2.1  py36h06ab557_0  conda-forge         
lsstdesc.coord                 1.2.1  py36h1cd168d_1  conda-forge         
lsstdesc.coord                 1.2.1  py36h60297b7_1  conda-forge         
lsstdesc.coord                 1.2.1  py36h863e41a_0  conda-forge         
lsstdesc.coord                 1.2.1  py36hadcdec8_1  conda-forge         
lsstdesc.coord                 1.2.1  py36he46ecad_1  conda-forge         
lsstdesc.coord                 1.2.1  py37h8ec247f_1  conda-forge         
lsstdesc.coord                 1.2.1  py37ha1cc60f_0  conda-forge         
lsstdesc.coord                 1.2.1  py37hb449ec0_1  conda-forge         
lsstdesc.coord                 1.2.1  py38h02bb52f_1  conda-forge         
lsstdesc.coord                 1.2.1  py38h9250c2b_1  conda-forge         
lsstdesc.coord                 1.2.1  py38ha0d09dd_0  conda-forge         
lsstdesc.coord                 1.2.1  py39hdbc963e_1  conda-forge         
lsstdesc.coord                 1.2.1  py39heea7e63_1  conda-forge         
lsstdesc.weaklensingdeblending             0.4          py27_0  conda-forge         
lsstdesc.weaklensingdeblending             0.4          py36_0  conda-forge         
lsstdesc.weaklensingdeblending             0.4          py37_0  conda-forge         
lsstdesc.weaklensingdeblending             0.4          py38_0  conda-forge         
lsstdesc.weaklensingdeblending             0.4            py_0  conda-forge         
lsstdesc.weaklensingdeblending             0.5  py36h9f0ad1d_0  conda-forge         
lsstdesc.weaklensingdeblending             0.5  py37hc8dfbb8_0  conda-forge         
lsstdesc.weaklensingdeblending             0.5  py38h32f6830_0  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py36h79c6626_1  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py36h9f0ad1d_0  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py37hc8dfbb8_0  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py37hf985489_1  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py38h32f6830_0  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py38h50d1736_1  conda-forge         
lsstdesc.weaklensingdeblending           1.0.0  py39h6e9494a_1  conda-forge 
```

**Comment by erykoff** [comment]
How do you list all the packages that are in `lsstdesc-env`?

**Comment by beckermr** [comment]
You look at the recipe.

**Comment by yymao** [comment]
BTW, [GCR](https://github.com/yymao/generic-catalog-reader) is now on conda-forge. Today I'll try to move GCRCatalogs there too. Based on the discussion here, I'll call it `desc.gcrcatalogs`?

**Comment by beckermr** [comment]
The name should be `lsstdesc-gcrcatalogs`.

**Comment by heather999** [comment]
I agree that would be easiest to keep track of. Guess there is the question of what is general versus what is specific to DESC.  I think code that lives in the LSSTDESC GH organization likely should always utilize the `lsstdesc-` prefix. Would we go back and request pyccl migrate to `lsstdesc-pyccl`?

**Comment by erykoff** [comment]
I don't know if that's always the case, but I'm not sure.  Right now `healsparse` is not `lsstdesc-healsparse` because although it's in the lsstdesc org, it's intended to be a general-purpose tool.  And I don't feel the need to have the verbose branding there.

**Comment by beckermr** [comment]
I agree. We leave this distinction up to the authors, like everything else.

**Comment by heather999** [comment]
So going back to Chris' question - what would we suggest to find a list of all "DESC" packages in conda-forge?  I just happen to be aware of what's available from experience, but that's not terribly useful for someone else. Asking users to look at a recipe may not be the most friendly option.  After install of something like lsstdesc-env, one can list the packages - I'm not aware of a way to get conda to list package dependencies before install (other than during the install).

**Comment by beckermr** [comment]
Here you go

```
$ mamba repoquery depends -c conda-forge lsstdesc-env

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.7.3) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Executing the query lsstdesc-env

conda-forge/osx-64       Using cache
conda-forge/noarch       Using cache


 Name                  Version      Build                       Channel           
───────────────────────────────────────────────────────────────────────────────────
 lsstdesc-env          2020.11.12.0 hd8ed1ab_0                  conda-forge/noarch
 numpy                 1.10.4       py27_blas_openblas_200      conda-forge/osx-64
 scipy                 0.17.1       np110py27_blas_openblas_200 conda-forge/osx-64
 pandas                0.17.0       np110py27_0                 conda-forge/osx-64
 matplotlib            1.5.3        np110py27_0                 conda-forge/osx-64
 h5py                  2.10.0       mpi_mpich_py27h36f4241_0    conda-forge/osx-64
 scikit-learn          0.17.1       np110py27_blas_openblas_200 conda-forge/osx-64
 dask                  0.10.0       py27_0                      conda-forge/osx-64
 numba                 0.36.1       py27hdc42b47_0              conda-forge/osx-64
 pytest                2.9.2        py27_0                      conda-forge/osx-64
 scikit-image          0.12.3       np110py27_0                 conda-forge/osx-64
 sphinx                1.4.5        py27_0                      conda-forge/osx-64
 pip                   10.0.1       py27_0                      conda-forge/osx-64
 seaborn               0.10.1       1                           conda-forge/osx-64
 astropy               1.3          np111py27_0                 conda-forge/osx-64
 tabulate              0.7.5        py27_0                      conda-forge/osx-64
 jupyter               1.0.0        py27_0                      conda-forge/osx-64
 bokeh                 0.12.10      py27_0                      conda-forge/osx-64
 psycopg2              2.6.1        py27_0                      conda-forge/osx-64
 pytables              3.2.3.1      np110py27_0                 conda-forge/osx-64
 google-cloud-bigquery 0.24.0       py27_0                      conda-forge/osx-64
 pyarrow               0.1.post     np111py27_6                 conda-forge/osx-64
 flake8                2.6.2        py27_0                      conda-forge/osx-64
 holoviews             1.8.0        py27_0                      conda-forge/osx-64
 corner                2.0.1        py27_0                      conda-forge/osx-64
 datashader            0.10.0       py_0                        conda-forge/noarch
 graphviz              2.38.0       0                           conda-forge/osx-64
 ipympl                0.0.10       py27_1                      conda-forge/osx-64
 nose                  1.3.7        py27_0                      conda-forge/osx-64
 cookiecutter          1.4.0        py27_1                      conda-forge/osx-64
 dask-jobqueue         0.1.1        py_0                        conda-forge/noarch
 conda                 4.1.11       py27_1                      conda-forge/osx-64
 emcee                 2.1.0        py27_0                      conda-forge/osx-64
 pybind11              1.7          py27_0                      conda-forge/osx-64
 python-snappy         0.5          py27_0                      conda-forge/osx-64
 healpy                1.10.3       np111py27_0                 conda-forge/osx-64
 ipdb                  0.10.1       py27_0                      conda-forge/osx-64
 pymssql               2.1.3.post16 py27_0                      conda-forge/osx-64
 sacc                  0.2.3        py_0                        conda-forge/noarch
 eigen                 3.2.8        0                           conda-forge/osx-64
 markupsafe            0.18         py27_0                      conda-forge/osx-64
 parsl                 0.7.2        py_0                        conda-forge/noarch
 fastparquet           0.0.1post2   py34_0                      conda-forge/osx-64
 astroplan             0.2.1        py27_0                      conda-forge/osx-64
 hdf5                  1.10.1       0                           conda-forge/osx-64
 yaml                  0.1.6        0                           conda-forge/osx-64
 cmake                 3.10.0       1                           conda-forge/osx-64
 fitsio                0.9.11       py27h3f622b2_1              conda-forge/osx-64
 healsparse            1.2.2        py_0                        conda-forge/noarch
 galsim                2.0.7        py27h04f5b5a_1000           conda-forge/osx-64
 pyccl                 2.0.1        py27h57a2de9_0              conda-forge/osx-64
 treecorr              3.3.11       py27h0a44026_1001           conda-forge/osx-64
 camb                  0.1.8.1      ha44fe06_0                  conda-forge/osx-64
 dask-mpi              1.0.1        py27_0                      conda-forge/osx-64
 dustmaps              1.0          py27_0                      conda-forge/osx-64
 extinction            0.2.2        np110py27_0                 conda-forge/osx-64
 gsl                   1.16         0                           conda-forge/osx-64
 namaster              0.9          py27h3b4356e_0              conda-forge/osx-64
 snappy                1.1.3        0                           conda-forge/osx-64
 python                3.8.6        hcfdab8c_0_cpython          conda-forge/osx-64

```

**Comment by beckermr** [comment]
The version weirdness is a bug: https://github.com/mamba-org/mamba/issues/618

**Comment by beckermr** [comment]
In terms of finding all DESC packages, there is not a great way to do this unless we force people into names which we should not and goes against our culture IMHO. We don't want to install everything at once anyway. If you need something, you'll know the name and then conda will install the dependencies.

**Comment by brianv0** [comment]
We could use this repo with a software-catalog.md or the like and just encourage people to add to that file when they add things to conda-forge. Of course, you can always search this github org and we can also recommend people add reference to packages (and their names) when they are added to conda-forge in the README.md
